### PR TITLE
Feature/pro 212 superfluid balance helpers

### PR DESCRIPTION
**Pull request by lukewickens1989**
- Added BalancesHelperV2 contract to get account balances for Superfluid tokens (pure and wrapped/native).
- Added test suite to check contract.
- Added 0.8.4 solidity compiler to hardhat.config.
- Updated solhint package to account for later Solidity code used in contract.

### CODE REVIEW

**Review by zincoshine — CHANGES_REQUESTED** [review]
Where is the deployment script for the new contract? Please have that ready and then test it locally or on Etherspot network.

File: package.json [hunk 1]
```diff
@@ -83,16 +87,18 @@
...
     "prettier": "1.19.1",
-    "solhint": "3.2.2",
+    "solhint": "^3.3.7",
     "solidity-coverage": "0.7.13",
     "ts-generator": "0.1.1",
     "ts-node": "8.10.2",
     "typechain": "6.0.3",
-    "typescript": "4.5.2"
+    "typescript": "4.5.2",
+    "web3": "^1.7.3"
```
[the remark this answers is not in the document]
Comment (lukewickens1989) [reply package.json:102]: This is a dev dependency of the [hardhat-web3 package](https://hardhat.org/plugins/nomiclabs-hardhat-web3). The reason this was used is that this is the method that Superfluid use in order to set up their tests ([Superfluid example code](https://github.com/superfluid-finance/protocol-monorepo/tree/dev/examples)), which I mainly used as a guide.

File: package.json [hunk 2]
```diff
@@ -67,9 +67,13 @@
     "verify:optimism": "hardhat verify-all --network optimism",
     "test": "hardhat test"
   },
-  "dependencies": {},
   "devDependencies": {
     "@nomiclabs/hardhat-etherscan": "2.1.7",
+    "@nomiclabs/hardhat-web3": "^2.0.0",
```
[the remark this answers is not in the document]
Comment (lukewickens1989) [reply package.json]: Done

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (zincoshine) [reply package.json]: Pls can we then change it to a specific web3 package? say 1.7.3 ?

[the remark this answers is not in the document]
Comment (zincoshine) [reply package.json]: @lukewickens1989

[the remark this answers is not in the document]
Comment (lukewickens1989) [reply package.json]: I think I locked the version in the last commit where I added the deployment script

**Review by zincoshine — APPROVED** [review]
LGTM

**Pull request merged**